clients/nodejs/validator-registration.js

```diff
@@ -0,0 +1,338 @@
+                'PrivateKey': signingKeyPair.privateKey.toHex(),
+            },
+            'VotingKey': {
+                'PublicKey': votingKeyPair.publicKey.toHex(),
+                'SecretKey': votingKeyPair.secretKey.toHex(),
+            }
+        };
+
+        // converting the JSON object to a string
+        const data = JSON.stringify(validatorConfiguration);
+
+        try {
+            fs.writeFileSync('validator-keys.json', data);
+        } catch (error) {
+            // logging the error
+            console.error(error);
+
+            throw error;
+        }
+
+        console.log('\n\nValidator configuration file sucessfully written.');
+        process.exit(0);
+    }
+
+    console.log(' Reading validator configuration.. ');
+    let validator_config = JSON5.parse(fs.readFileSync(argv.validator));
+
+    console.log(validator_config);
+
+    Nimiq.Log.i(TAG, `Nimiq NodeJS Client starting (network=${config.network}`
+        + `, ${config.host ? `host=${config.host}, port=${config.port}` : 'dumb'}`);
+
+    Nimiq.GenesisConfig.init(Nimiq.GenesisConfig.CONFIGS[config.network]);
+
+    for (const seedPeer of config.seedPeers) {
+        let address;
+        switch (seedPeer.protocol) {
+            case 'ws':
+                address = Nimiq.WsPeerAddress.seed(seedPeer.host, seedPeer.port, seedPeer.publicKey);
+                break;
+            case 'wss':
+            default:
+                address = Nimiq.WssPeerAddress.seed(seedPeer.host, seedPeer.port, seedPeer.publicKey);
+                break;
+        }
+        Nimiq.GenesisConfig.SEED_PEERS.push(address);
+    }
+
+    const clientConfigBuilder = Nimiq.Client.Configuration.builder();
+    clientConfigBuilder.protocol(config.protocol, config.host, config.port, config.tls.key, config.tls.cert);
+    if (config.reverseProxy.enabled) clientConfigBuilder.reverseProxy(config.reverseProxy.port, config.reverseProxy.header, ...config.reverseProxy.addresses);
+    if (config.passive) clientConfigBuilder.feature(Nimiq.Client.Feature.PASSIVE);
+    clientConfigBuilder.feature(Nimiq.Client.Feature.MEMPOOL);
+
+    const clientConfig = clientConfigBuilder.build();
+    const networkConfig = clientConfig.networkConfig;
+
+    $.consensus = await (!config.volatile
+        ? Nimiq.Consensus.nano(networkConfig)
+        : Nimiq.Consensus.volatileNano(networkConfig));
+
+    $.client = new Nimiq.Client(clientConfig, $.consensus);
+    $.blockchain = $.consensus.blockchain;
+    $.accounts = $.blockchain.accounts;
+    $.mempool = $.consensus.mempool;
+    $.network = $.consensus.network;
+
+    Nimiq.Log.i(TAG, `Peer address: ${networkConfig.peerAddress.toString()} - public key: ${networkConfig.keyPair.publicKey.toHex()}`);
+
+    const chainHeight = await $.client.getHeadHeight();
+    const chainHeadHash = await $.client.getHeadHash();
+    Nimiq.Log.i(TAG, `Blockchain state: height=${chainHeight}, headHash=${chainHeadHash}`);
```

Comment:
Without waiting for consensus (e.g. with `await $.client.waitForConsensusEstablished()`), this will report height 1 and the genesis hash.